presto-main/src/main/java/com/facebook/presto/operator/aggregation/builder/SpillableHashAggregationBuilder.java

```diff
@@ -221,7 +227,7 @@ public void close()
 
         hashAggregationBuilder.setOutputPartial();
 
-        Iterator<Page> mergedSpilledPages = MergeHashSort.merge(
+        Iterator<Page> mergedSpilledPages = mergeHashSort.get().merge(
```

Comment:
I don't like this. I moved the identical assignment out of those two private method before the if-else where those methods are called to avoid redundancy etc, so I don't like introducing it again with this check. I'd rather keep the original duplicate in-place assignment then. Reverting to that.